exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -134,6 +140,45 @@ type RosaMachinePoolAutoScaling struct {
 	MaxReplicas int `json:"maxReplicas,omitempty"`
 }
 
+// RosaMachinePoolUpgradeConfig specifies upgrade management configuration which set MaxUnavailable & MaxSurge number of nodes during upgrade.
+type RosaMachinePoolUpgradeConfig struct {
+	// MaxUnavailable is the maximum number of nodes that can be unavailable during the update.
+	// Value can be an absolute number (ex: 5) or a percentage of desired nodes (ex: 10%).
+	// Absolute number is calculated from percentage by rounding down.
+	//
+	// MaxUnavailable can not be 0 if MaxSurge is 0 and defaults to 0.
+	//
+	// Example: when MaxUnavailable is set to 30%, old nodes can be deleted down to 70% of
+	// desired nodes immediately when the rolling update starts. Once new nodes
+	// are ready, more old nodes be deleted, followed by provisioning new nodes,
+	// ensuring that the total number of nodes available at all times during the
+	// update is at least 70% of desired nodes.
+	//
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:default=0
```

Comment:
I don't think we need to provide defaults. OCM might have different values or might change them anytime.

Reply:
If OCM will change that it is API changes and we need to change it as well. Those default values same as the Hypershift default values and we have to expose it to the user.

Reply:
> If OCM will change that it is API changes and we need to change it as well.

that's the point, if we don't set defaults in the client side, then we don't have to change them when the backend(OCM) change theirs. If the fields are not set explicitly by the user, we send empty values to OCM so they can set the appropriate defaults.
Otherwise, we will end up in the same situation as the `autoRepair`